### DISCUSSION TITLE
How to properly make package unthrow from @joemarshall with new pyodide

### DISCUSSION THREAD

**Discussion by tjefferson401** [pyodide/pyodide#2605] [General]
Unthrow has been mentioned several times as one way to handle sync IO. I used an old build of unthrow to test its functionality, and it worked, however I can't seem to build it with Pyodide stable release. I'm wondering if anyone has used it with a recent version of pyodide.
I'm using the meta.yaml and setup.py from the original repo which I will link below:
https://github.com/joemarshall/unthrow

The issue I am running up against is when building with `python -m pyodide_build buildpkg meta.yaml`, I receive an error 👍   File ```
"setup.py", line 2, in <module>
    from Cython.Build import cythonize
ModuleNotFoundError: No module named 'Cython'
```

I understand Cython is not supported by pyodide.
Any suggestions?

**Comment by hoodmane** [comment]
Cython is supported, it is used by many packages we do support. You may need to add:

```yaml
build:
  script: |
    pip install -t $HOSTSITEPACKAGES cython
```
see for instance `packages/numcodecs/meta.yaml`.

But `unthrow` would need to be updated, it only supports Python 3.8. It says: "3.9 stores stack level as a number not a pointer, it will need this minor change to work on 3.9" I'm not sure what needs to be changed to work with Python 3.10.

**Reply by tjefferson401** [reply]
Thanks for the clarification! Currently using V0.19.1 of pyodide and running up against "fatal error: 'Python.h' file not found" on the same command. I'm using the docker img and am wondering if there's a fix for this. I've checked the issues and I haven't seen one come up anywhere. Thanks again.

Edit: Using 0.19.1 docker img and python version is 3.9.5

**Reply by hoodmane** [reply]
You have to run `make -C cpython` first. (We are well on our way to not needing to build Python to build packages, hopefully by v0.22 this will be possible.)

**Reply by tjefferson401** [reply]
Sorry to continue, but I run this at what level?

**Reply by hoodmane** [reply]
In the docker image inside of `/src/` (absolute path). You can also run `make` inside `/src/cpython/`.

**Reply by tjefferson401** [reply]
Thank you for being so kind with your time.
